### PR TITLE
Code cleanup and method docs

### PR DESCRIPTION
**Pull request by Nyholm**
This PR cleans up the Authentication API. It adds docs and make sure the metod parameters make sense. It make sure all methods are camelCase

#### These functions has changed name: 
get_authorize_link -> getAuthorizeLink
get_samlp_link -> getSamlpLink
get_samlp_metadata_link -> getSamlpMetadataLink
get_wsfed_link -> getWsfedLink
get_wsfed_metadata_link -> getWsfedMetadataLink
get_logout_link -> getLogoutLink
authorize_with_accesstoken -> authorizeWithAccessToken
email_passwordless_start -> emailPasswordlessStart
sms_passwordless_start -> smsPasswordlessStart
userinfo -> userInfo
oauth_token -> oauthToken (now also private)
code_exchange -> codeExchange
login_with_default_directory -> loginWithDefaultDirectory
client_credentials -> clientCredentials
dbconnections_signup -> dbconnectionsSignup
dbconnections_change_password -> dbconnectionsChangePassword

#### These functions has changes signature: 
* New signature for logoauthTokenn function: `oauthToken($grantType, array $options)`
* New signature for login function: `login($username, $password, $realm, array $options = [])`
* New signature for loginWithDefaultDirectory: `loginWithDefaultDirectory($username, $password, array $options = [])`

### CODE REVIEW

File: src/API/Authentication.php [hunk 1]
```diff
@@ -170,46 +217,57 @@ public function sms_passwordless_start($phone_number)
...
-     * @param {Object} options.client_id
-     * @param {Object} options.client_secret [optional] Only if grant type: client_credentials
-     * @param {Object} options.username  [optional] Only if grant type: password/password-realm
-     * @param {Object} options.password  [optional] Only if grant type: password/password-realm
-     * @param {Object} options.scope     [optional]
-     * @param {Object} options.audience  [optional]
+     * @return array
      */
-    public function oauth_token($options = [])
+    public function oauthToken($grantType, array $options)
```
Comment (cocojoe) [inline src/API/Authentication.php]: Should this become a private method, it seems to mostly be a wrapper now @glena ?

File: src/API/Authentication.php [hunk 2]
```diff
@@ -231,120 +287,121 @@ public function oauth_token($options = [])
...
-     * @param {Object} options:
-     * @param {Object} options.username
-     * @param {Object} options.password
-     * @param {Object} options.realm
-     * @param {Object} options.scope [optional]
-     * @param {Object} options.audience [optional]
+     * @return array
      */
-    public function login($options)
+    public function login($username, $password, $realm, array $options = [])
```
Comment (cocojoe) [inline src/API/Authentication.php]: So perhaps we should use this opportunity to promote some options to params, such as `scope`  and `audience`?

**Comment by cocojoe** [comment]
Oh also liking the coding throw back to `C` style `sprintf` 😄

File: src/API/Authentication.php (hunk 1 above)
Comment (Nyholm) [inline src/API/Authentication.php]: I guess you are correct. Ive fixed this.

**Comment by Nyholm** [comment]
> So perhaps we should use this opportunity to promote some options to params, such as scope and audience?

I agree this might be good. I suggest to make that to a separate PR. This PR is massive and hard to review anyways. Putting more changes/features in it will make it more difficult. 

Im happy to make such PR and discuss pros and cons as soon as this get merged.

**Comment by cocojoe** [comment]
OK let's look at prompting params in a new PR, we are introducing a lot of `changes`. However it makes sense in terms of adopting PSR community standards and moving the SDK forward.

**Comment by Nyholm** [comment]
Thank you for merging.

**Pull request merged**